File: app/src/main/java/social/entourage/android/view/CountryCodePicker/Country.java
package social.entourage.android.view.CountryCodePicker;

public class Country {
    private String iso;
    private String phoneCode;
    private String name;

    public Country(String iso, String phoneCode, String name) {
        this.iso = iso;
        this.phoneCode = phoneCode;
        this.name = name;
    }

    public String getIso() {
        return iso;
    }

    public void setIso(String iso) {
        this.iso = iso;
    }

    public String getPhoneCode() {
        return phoneCode;
    }

    public void setPhoneCode(String phoneCode) {
        this.phoneCode = phoneCode;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    /**
     * If country have query word in name or name code or phone code, this will return true.
     */
    boolean isEligibleForQuery(String query) {
        query = query.toLowerCase();
        return getName().toLowerCase().contains(query)
                || getIso().toLowerCase().contains(query)
                || getPhoneCode().toLowerCase().contains(query);
    }
}

File: app/src/main/java/social/entourage/android/tools/view/countrycodepicker/Country.kt
package social.entourage.android.tools.view.countrycodepicker

import java.util.*

class Country(var iso: String, var phoneCode: String, var name: String) {

    /**
     * If country have query word in name or name code or phone code, this will return true.
     */
    fun isEligibleForQuery(query: String): Boolean {
        val queryLowerCase = query.lowercase(Locale.getDefault())
        return (name.lowercase(Locale.getDefault()).contains(queryLowerCase)
                || iso.lowercase(Locale.getDefault()).contains(queryLowerCase)
                || phoneCode.lowercase(Locale.getDefault()).contains(queryLowerCase))
    }

}